File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
class BirdCount(int[] birdsPerDay)
{
    private readonly int[] birdsPerDay = birdsPerDay;

    public static int[] LastWeek() => [0, 2, 5, 3, 7, 8, 4];

    public int Today() => birdsPerDay[^1];

    public void IncrementTodaysCount() => birdsPerDay[^1]++;

    public bool HasDayWithoutBirds() => birdsPerDay.Any(count => count == 0);

    public int CountForFirstDays(int numberOfDays) => birdsPerDay.Take(numberOfDays).Sum();

    public int BusyDays() => birdsPerDay.Count(count => count >= 5);
}

File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

import java.util.Arrays;
import java.util.stream.IntStream;

class BirdWatcher {
    private static final int[] LAST_WEEK = {0, 2, 5, 3, 7, 8, 4};
    private static final int BUSY_DAY_THRESHOLD = 5;

    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public static int[] getLastWeek() {
        return LAST_WEEK.clone();
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length - 1]++;
    }

    public boolean hasDayWithoutBirds() {
        return birds().anyMatch(count -> count == 0);
    }

    public int getCountForFirstDays(int numberOfDays) {
        return birds().limit(numberOfDays).sum();
    }

    public int getBusyDays() {
        return (int) birds().filter(count -> count >= BUSY_DAY_THRESHOLD).count();
    }

    private IntStream birds() {
        return Arrays.stream(birdsPerDay);
    }
}
